core/define/src/mill/define/internal/WatchSig.scala

```diff
@@ -0,0 +1,22 @@
+package mill.define.internal
```

Comment:
Why does this stuff need to be moved to `mill.define`? AFAIK we only use it in the `runner` codebase which is not exposed to the `build.mill` classpath that `mill.define` is meant for

Reply:
I added this here because I need to use some of it from `MillBuildServer`. See the explanation about concurrency below for more details.